unblob/processing.py

```diff
@@ -112,6 +118,30 @@ def get_extract_dir_for(self, path: Path) -> Path:
         return extract_dir.expanduser().resolve()
 
 
+def sandbox(extract_dir: Path, report_file: Optional[Path]):
+    restrictions = [
+        AccessFS.read("/"),
+        AccessFS.read_write("/dev/shm"),  # noqa: S108
+        AccessFS.read_write(extract_dir.as_posix()),
+        AccessFS.make_dir(extract_dir.parent.as_posix()),
+    ]
+
+    if report_file:
+        restrictions += [
+            AccessFS.read_write(report_file),
+            AccessFS.make_reg(report_file.parent),
+        ]
+
+    if "pytest" in sys.modules:
+        restrictions += [
+            AccessFS.read_write("/tmp"),  # noqa: S108
+            AccessFS.read_write("/build"),
+            AccessFS.read_write(Path(__file__).parent.parent.resolve().as_posix()),
+        ]
```

Comment:
restrictions could be part of `ExtractionConfig` we pass for process_file, so test and report specific values could be set-up in fixture and command line processing.

Reply:
I like your thinking.

Reply:
Working on it

Reply:
gave it a try yesterday but failed :)

Reply:
I found an interesting thing: the only test, that actually calls the cli and does a full extraction is the `tests/test_cli.py::test_skip_extension` case, everything else works with a mocked process_file :)